### ISSUE TITLE
Next bugfix release

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#7353]
After publishing 2019.02.23 we do now have some changes on branch `release` which might worth to be released soon. I will create a new release candidate maybe tomorrow to give it a test run on our beta channels.

If there is any other bugfix for `release` branch currently worked on, please let me know here, so we could decide to either include it or wait until the next bugfix release after the upcoming.

**Comment by moving-bits** [comment]
I'm currently preparing a small PR to disable static maps loading.

**Comment by Lineflyer** [comment]
Thanks @moving-bits PR#7357 is merged. Will smoke test in another RC but not yet publish it to the beta channel.

**Comment by Lineflyer** [comment]
Found another blocking issue #7358 while smoke testing the RC version.

**Comment by Lineflyer** [comment]
As #7358 was solved thanks to @moving-bits I will generate and smoke test a new RC.

**Comment by Lineflyer** [comment]
Beta version 2019.03.15-RC is out on the beta channel:

Beta version: 
- Fix: App crashing when trying to add log image from camera - Fix: Button to change map type not visible 
- Fix: Coordinate input field broken in DMS format 
- Fix: Action buttons on list select dialog might be not be completely visible on some phones 
- Removed: Support for Android 4.0.x devices 
- Fix: Disable static map loading by default as Google static maps are no longer available

**Comment by moving-bits** [comment]
>     * Fix: Disable static map loading by default as Google static maps are no longer available

Just for clarification: Static maps loading is not only disabled by default in this version, it's disabled and cannot be activated by the user any more - so from a user's point of view it's removed as well.
(The actual technical removal of static maps functions is done by PR 7375 which IMHO we should merge _after_ release of this version to testdrive it for unwanted side effects first.)

**Comment by Lineflyer** [comment]
> Just for clarification: Static maps loading is not only disabled[...]

Yes, I know. I wanted to make it simple for the change log.

**Comment by Lineflyer** [comment]
FYI:
2019.03.17-RC has just been published to the beta channel.

New compared to last Beta is the possible fix for logging timeouts #7388

**Comment by Lineflyer** [comment]
The beta seems to perform normal looking at Google Play Console.

There is still the rather high amount of NPE crashes, which are already listed here as #7303 and #7300 .
Unless someone contributes a fix for this, I suggest to release rather soon this week, to get some bugs solved in the field.

Someone could also load strings from Crowdin again. There was quite some activity after the last upload some days ago.

My plan would be to release Wednesday, March 27th.

**Comment by rsudev** [comment]
I plan to import strings this evening. Regarding the map.onResume-crashes: as mentioned I have a (not complicated) idea for a fix, but without a reproducible scenario, everything is just a wild guess...

**Comment by Lineflyer** [comment]
> Regarding the map.onResume-crashes: as mentioned I have a (not complicated) idea for a fix, but without a reproducible scenario, everything is just a wild guess...

Could this be the same problem as when using the widget to the live map (#7391). In that case you would have a reproducible scenario.

**Comment by Lineflyer** [comment]
Otherwise we could also try out your fix in a beta version. It will be pretty obvious in the play console crash feedback whether your fix is working then.

**Comment by rsudev** [comment]
Than I will aim for a fix for the widget issue and see whether this aligns with my ideas regarding the other topic.

**Comment by moving-bits** [comment]
Besides including a possible fix for the NPE or widget issue I would suggest to include #7410 only from the open PR, as it is the only open bug fix PR. All other PR are either new features or code cleanup, which could (and should) be merged into master shortly after the new release IMHO.

**Comment by bekuno** [comment]
@rsudev The string "cache_menu_navigon" is still contained in 4 strings.xml. (es, ro, sl, ceb). Please check it on the next import.

**Comment by Lineflyer** [comment]
> Besides including a possible fix for the NPE or widget issue I would suggest to include #7410 only from the open PR, as it is the only open bug fix PR. 

The feature itself is only contained on `master` at the moment. Therefore merging this fix is independently from release planning as we only talk about the `release` branch here.

After this next bugfix release we still can decide whether we keep `release` for another bugfix-only release or merge from `master` to get a shorter cycle of releasing new features. I would be in favor of the second option as `master` looks rather stable at the moment.

**Comment by moving-bits** [comment]
Oh, I thought we were talking about pushing the `master` branch to market as next release (which I would favour, since it's already a long time since the last release and there are a couple new features). (But you're right, this issue was about the `release` branch in the beginning.)

You say `master` branch seems to be pretty stable at the moment. Any reasons why not to release `master` branch?

**Comment by Lineflyer** [comment]
> You say master branch seems to be pretty stable at the moment. Any reasons why not to release master branch?

This is more or less the process I used to follow last years which has proven to allow enough time for testing with a limited amount of changes. Furthermore I always try to keep the `release` branch in a state, which allows us to release on short term in case of need (e.g. website change) without (significant) regressions.

At this point of time the release process from `release`branch (which I call "bugfix release") has already been started by pushing a beta version to Google Play a week ago.

Regarding the merging from `master` to `release` my process is a bit based on my feeling about the quality and stability of the nightlies combined with the commit history on `master`. As soon as I see a good chance to branch a stable new feature release I should out loudly here :) As stated above, I would like to do so perhaps immediately after the upcoming release.

**Comment by Lineflyer** [comment]
Addition:
But feel free to continue implementation of features (and following fixes for them) on `master` until then.

**Comment by moving-bits** [comment]
Ok, thanks for the update & clarification.

For me with hotfixes on `release` branch I associated a timespan of a couple of days after a release version, whereas a regular release cycle (`master` branch) would be like every month (both only if changes are available) - that's why I thought releasing a feature version would be next. (But maybe that cycle is wishful thinking from my side and too ambitious.)

**Comment by rsudev** [comment]
@bekuno I'll do (likely tomorrow) 
I have a fix for the widget issue in the pipeline, I'll try to push it this evening.

**Comment by Lineflyer** [comment]
>  (But maybe that cycle is wishful thinking from my side and too ambitious.)

It strongly depends on the level of activity in this project. Before the rather large gap due to SDK-26 change, we used to have a feature release every 2-3 months and a bugfix release more regular in case of need.
Bugfix release is controlled by external demand most of the time (e.g. hot fix just after a new feature release, urgent fix due to website change)where as the feature release cycle is mainly suggested by my/our testing results and stability experience of `master`.

My past experience has shown, that shorter cycles are easier to handle as the amount of changes is not so huge, however it also depends on my availability to test and others availability to fix what we found during test :)
For this year I want to work towards merging `master` to `release` as often as possible (smaller cycles, less changes per release). Lets see how it goes.

**Comment by Lineflyer** [comment]
I published a new beta version 2019.03.30-RC1 including the fix for #7391 to see whether it also fixes #7303 and/or #7300.
Will report back after the weekend.

**Comment by Lineflyer** [comment]
@rsudev Quick feedback from beta:

Seems those NPE are no longer occuring, but I do see some remaining NPE (now that the large amount of NPE from map are fixed) in EditWaypointActivity and CompassActivity for onResume. 
Seems this needs to be handled in all acitivites?

**Comment by Lineflyer** [comment]
Created #7442 and #7443 for the NPE mentioned above.
@rsudev Any chance you tackle these similar to your fix for #7391 on short term? Then we should wait with the next release until then.

**Comment by Lineflyer** [comment]
FYI:

A new RC version has been smoke tested and now published to the Google Plaz beta channel (2019.04.03-RC)
It contains the latest fixes from @rsudev regarding the "OnResume" crashes.

Will report back how it performs in 1-2 days

**Comment by Lineflyer** [comment]
The new RC version 2019.04.03-RC is behaving normal.

We had approx. 100 users live in parallel. Google Play shows no sign of the previous "OnResume" crashes and also no new/unknown other crashes/problems.

I suggest to release a new version tonight, please object until then if you see other open points.

Thereafter I will open a new thread to discuss, whether we shall merge `master` to `release` to have the next following release as a feature release (which I currently prefer).

**Comment by Lineflyer** [comment]
Starting release process:


- [ ] Check/Import incoming Translations from Crowdin
- [ ] Check/Insert correct version name as headline in ```changelog-release.xml```
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid 
- [ ] Rename/Create milestones on Github
- [ ] Trigger notification to status.cgeo.org via CI server once the release is available on Google Play
- [ ] Post release info to Facebook 
- [ ] Post release info to G+ 
- [ ] Post release info to Twitter

**Comment by Lineflyer** [comment]
Note to myself and team:
I will skip and remove the process to use milestones for release marking...it just creates no additional information as we have all of that in the version management / commit history.

**Comment by Lineflyer** [comment]
Well, theoretically all is done for release, but seems Google Play needs a looong time to digest the new release. It has still not confirmed it and no active installations to see on status.

I will check back later to finalize this with notification, etc.

**Comment by Lineflyer** [comment]
We are done. Thanks for your contributions for this release :)
